assets/src/scripts/components/map/floodDataSelector.js

```diff
@@ -0,0 +1,55 @@
+const { createSelector } = require('reselect');
+
+
+export const floodStagesSelector = state => state.floodData.stages;
+
+export const floodExtentSelector = state => state.floodData.extent;
+
+export const floodGageHeightSelector = state => state.floodState.gageHeight;
+
+/*
+ * Provides a function which returns True if flood data is not empty.
+ */
+export const hasFloodDataSelector = createSelector(
+    floodStagesSelector,
+    (stages) => stages.length > 0
+);
+
+/*
+ * Provides a function which returns the stage closest to the gageHeight
+ */
+export const floodStageHeightSelector = createSelector(
+    floodStagesSelector,
+    floodGageHeightSelector,
+    (stages, gageHeight) => {
+        let result = null;
+        if (stages.length && gageHeight) {
+            result = stages[0];
+            let diff = Math.abs(gageHeight - result);
+            stages.forEach((stage) => {
+                let newDiff = Math.abs(gageHeight - stage);
+                if (newDiff <= diff) {
+                    diff = newDiff;
+                    result = stage;
+                }
+            });
+        }
```

Comment:
I've been pondering this function... do you think it'd be more efficient to do this to be the min?
```
result = stages.filter(i => Math.abs(gageHeight-i) == Math.min(...stages.map(i => Math.abs(gageHeight-i))))[0]
```

Reply:
That's a great idea! Thanks

Reply:
On second glance, I think I'll leave as is. I think that function is no more efficient although it is more concise. I found it a little hard to reason about.